modules/outline-of-a-basic-values-file.adoc

```diff
@@ -0,0 +1,17 @@
+:_content-type: CONCEPT
+:imagesdir: ../../images
+
+[id="outline-of-a-basic-values-file"]
+= Outline of a basic values file
+
+A values file typically consists of three main sections:
```

Comment:
In a values file, every variable declares an element (with the default value for that element in the values file) that can be used in helm templates. And then every file (or override) that is added to the rendering pipeline will then override that default, so that the last value in, wins.

The three-fold distinction is a specific application of the helm override scheme for validated patterns. (We haven't done a great job ourselves of distinguishing between "main" and "global" and we effectively use those terms interchangeably. There's a subtle difference in helm (specifically: "global" variables are also available in subcharts; so if the chart declares dependent charts the "global" values from the top-level chart will also be passed in to the subcharts to render them).

The "clusterGroup" is a definition clusters that are expected to have a nearly-identical configuration amongst the members of the clustergroup since they all serve the same architectural purpose in a design. A clustergroup may have one or many members; there is a main clustergroup (defined, by convention, in main.clusterGroupName in values-global.yaml) that is the "hub" cluster; this is the "starting point" for each pattern. In addition, additional clusterGroups may be defined that are managed by the main clusterGroup (through RHACM). So what you're saying about a clusterGroup is basically true, but that comes from the way that patterns use values files, rather than from an inherent quality in values files. (his is because clusterGroup is a specific parameter to the chart that defines the validated patterns framework.

Generally speaking, values files are free-form YAML and are free to use any of the YAML variable types.